src/System.Globalization/tests/CompareInfo/CompareInfoTests.IndexOf.cs

```diff
@@ -55,6 +55,7 @@ public static IEnumerable<object[]> IndexOf_TestData()
             yield return new object[] { s_invariantCompare, "Exhibit \u00C0", "a\u0300", 0, 9, CompareOptions.OrdinalIgnoreCase, -1 };
             yield return new object[] { s_invariantCompare, "FooBar", "Foo\u0400Bar", 0, 6, CompareOptions.Ordinal, -1 };
             yield return new object[] { s_invariantCompare, "TestFooBA\u0300R", "FooB\u00C0R", 0, 11, CompareOptions.IgnoreNonSpace, 4 };
+            yield return new object[] { s_invariantCompare, "o\u0308", "o", 0, 2, CompareOptions.None, -1 };
```

Comment:
Could you please add a test cases for (IndexOf, IsPrefix, IsSuffix) validating the following case? 

```C#
 Console.WriteLine($"{CultureInfo.GetCultureInfo("sk-SK").CompareInfo.IsSuffix("ch", "h")}"); // print False
```
I know you code shouldn't go through your optimization (as it is not en culture) but would help to ensure that too. 

Thanks  a lot for your fixes here.